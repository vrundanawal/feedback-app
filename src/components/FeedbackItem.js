import React from "react";
import { FaTimes } from "react-icons/fa";
import Card from "./shared/Card";
import PropTypes from "prop-types";

const FeedbackItem = ({ item, handleDelete }) => {
  return (
    <Card>
      <div className="num-display">{item.rating}</div>
      <button
        data-testid="delete-list"
        className="close"
        onClick={() => handleDelete(item.id)}
      >
        <FaTimes color="purple" />
      </button>
      <div className="text-display">{item.text}</div>
    </Card>
  );
};

FeedbackItem.propTypes = {
  item: PropTypes.object.isRequired,
};

export default FeedbackItem;
