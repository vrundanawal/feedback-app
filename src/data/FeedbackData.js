const FeedbackData = [
  {
    id: 1,
    rating: 10,
    text: "Using context Api Item 1.",
  },
  {
    id: 2,
    rating: 9,
    text: "Using context Api Item 2",
  },
  {
    id: 3,
    rating: 8,
    text: "Using context Api Item 3",
  },
  {
    id: 4,
    rating: 9,
    text: "Using context Api Item 3",
  },
];

export default FeedbackData;
