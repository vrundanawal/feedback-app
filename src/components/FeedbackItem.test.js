import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import FeedbackData from "../data/FeedbackData";
import FeedbackItem from "./FeedbackItem";
import { shallow } from "enzyme";

//import userEvent from "@testing-library/user-event";

describe("<FeedbackItem/>", () => {
  it("should render the feedback List component", () => {
    expect(FeedbackItem).toBeTruthy();
  });

  it("should have delete button", () => {
    // const deleteBtn = screen.getByTestId("delete-list");
    // expect(deleteBtn).toBeInTheDocument();
    render(<FeedbackItem item={FeedbackData[0]} />);
    const deleteBtn = screen.getByTestId("delete-list");
    expect(deleteBtn).toBeInTheDocument();
    //expect(screen.queryByTestId("delete-list")).not.toBeInTheDocument();
  });
});

test("should test feedbackItem component", () => {
  const wrapper = shallow(<FeedbackItem item={FeedbackData[0]} />);
  expect(wrapper).toMatchSnapshot();
});

//It should remove the item from the list
test("<FeedbackList/>", () => {
  const handleDelete = jest.fn();
  const item = {
    id: 12,
    rating: 5,
    text: "Lorem ipsum dolor sit amet consectetur adipisicing elit.",
  };
  const { getByTestId } = render(<FeedbackItem item={item} />);
  const deleteBtn = getByTestId("delete-list");
  fireEvent.click(deleteBtn);
  // expect(handleDelete).toHaveBeenCalledTimes(1);
  expect(handleDelete).toBeCalledWith(item.id);
});
