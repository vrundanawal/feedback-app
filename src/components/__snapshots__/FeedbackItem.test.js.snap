// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test feedbackItem component 1`] = `
<Card
  reverse={false}
>
  <div
    className="num-display"
  >
    10
  </div>
  <button
    className="close"
    data-testid="delete-list"
    onClick={[Function]}
  >
    <FaTimes
      color="purple"
    />
  </button>
  <div
    className="text-display"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. consequuntur vel vitae commodi alias voluptatem est voluptatum ipsa quae.
  </div>
</Card>
`;
